### PR TITLE
move documentation from txt to rst files

### PR DESCRIPTION
**Pull request by gaborbernat**
Currently the doc files for some reason are named txt files, which is kinda inconvenient as many editors don't know how to highlight them; here I've renamed it to rst as these are in fact reStructuredText files, to reflect that 👍

### CODE REVIEW

**Comment by gaborbernat** [comment]
a small change, made the change to CHANGELOG file too 👍

**Comment by obestwalter** [comment]
Thanks @gaborbernat that you tackle all these things at the moment. I will try to get 2.8 out of the way until the end of the week and then we can make a bit mergefest.

**Comment by obestwalter** [comment]
@gaborbernat - sorry this one went a bit stale. Could you integrate master into your PR? Then we can merge it.

**Comment by gaborbernat** [comment]
@obestwalter done it, and a lot more; I've purposefully broke it down to multiple commits I can do it one PR at a time if you need it.

here's a list of changes:

- move documentation from txt to rst files
- remove announce data as this is already present now in changelog (we sort of duplicate it otherwise the information, as it's available in two places)
- move to tox sphinx-build command from pytest, so the documentation can be explored afterwards, and threat warnings as errors, so they are not ignored
- fix links that no longer exist or redirect permanently
- make literals code blocks with appropriate syntax highlighting
- remove generated commented lines from conf.py
- add ``SSL_CERT_FILE`` env variable sphinx-builds ``tls_cacerts`` so that it will work behind corporate proxies

**Comment by gaborbernat** [comment]
one more thing we might want to do is to somehow add line length check for rst files :+1: although not sure what's the best method for that; otherwise the line length seems arbitrary in documentation files

**Comment by obestwalter** [comment]
Wow! Great! I'm a bit under the weather at the moment, but I will definitely have a look tomorrow. Things are moving along nicely :)

I have also been eying up the pytest docs theme. @nicoddemus would there be objections to use that for the tox docs?

**Comment by nicoddemus** [comment]
Not from my part, but it might be better to consult @hpk42's opinions on this (he might have an idea for a different theme for tox).

Also you guys might take a look some time at the [documentation-restructure branch](https://docs.pytest.org/en/documentation-restructure/) which @evildmp has started and I think is looking very promising.

File: doc/announce/changelog-only.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Less announcing, more change-logging
+------------------------------------
+
+With version 2.5.0 we dropped creating special announcement documents and rely on communicating
+all relevant changes through the
+`CHANGELOG <https://github.com/tox-dev/tox/blob/master/CHANGELOG>`_. See at
+`pypi <https://pypi.org/project/tox/>`_ for a rendered version of the last changes containing
+links to the important issues and pull requests that were integrated into the release.
+
```
Comment (obestwalter) [inline doc/announce/changelog-only.rst]: When we really get rid of them I would add the following also:

The historic release announcements are still online here:

* http://tox.readthedocs.io/en/2.4.0/announce/release-0.5.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.0.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.1.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.2.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.3.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.4.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.4.3.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.8.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-1.9.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-2.0.html
* http://tox.readthedocs.io/en/2.4.0/announce/release-2.4.0.html

**Comment by gaborbernat** [comment]
@obestwalter added

**Comment by obestwalter** [comment]
I am just fixing up that coverage stuff (found out how to work aorund pytest-cov bug) and will merge then.

**Pull request merged**